### PR TITLE
Add url origin to config.yaml

### PR DESCRIPTION
**Pull request by Keyology**
@deliciousmonster Please let me know if this is what you would like added to the main branch

### CODE REVIEW

**Comment by deliciousmonster** [comment]
I am not sure you need to use fs and that yaml package… I believe you can access the variables similarly to the way you access tables.

@kylebernhardy - am I imagining this?

**Comment by kylebernhardy** [comment]
> I am not sure you need to use fs and that yaml package… I believe you can access the variables similarly to the way you access tables.
> 
> @kylebernhardy - am I imagining this?

In our `server` global you can access `server.config` but that resolves to the JSON of the `harperdb-config.yaml`.  I haven't seen in our docs any mention of accessing the application's `config.yaml` but that doesn't mean it's not available, just hasn't been put in docs potentially. @Keyology have you tried debugging your app and inspecting the global `server` and `this` to see what attributes are available?

Currently this is what we document for globals: https://docs.harperdb.io/docs/technical-details/reference/globals#server

**Comment by Keyology** [comment]
@deliciousmonster  I initially tried importing the variable similar to how tables are imported as you described; however, I encountered an error because the variable couldn’t be read. After doing some research, I found that in Node.js and Fastify, YAML files aren't automatically parsed or imported like JavaScript modules. To my understanding, you need to read and parse the YAML file explicitly, as this is not supported by default. I can look into importing from harperdb-config.yaml and the server.config file. @kylebernhardy

**Comment by deliciousmonster** [comment]
@Keyology If it has to be in the harperdb-config.yaml, let's do that. Ideally, our template components should have as few dependencies as possible to make it as easy as possible to see the results...

**Comment by kylebernhardy** [comment]
> @Keyology If it has to be in the harperdb-config.yaml, let's do that. Ideally, our template components should have as few dependencies as possible to make it as easy as possible to see the results...

Let's not add app specific setting in the `harperdb-config.yml` connect with @kriszyp to find out how to get access to the application's `config.yaml`

**Comment by Keyology** [comment]
@kylebernhardy I synced with Kris via Slack, and he said the origin URL cannot be imported into the resources.js file from the config.yaml file.  @deliciousmonster

File: config.yaml
Comment (deliciousmonster) [inline config.yaml:1]: let's make this a valid url

File: resources.js
Comment (deliciousmonster) [inline resources.js:1]: you should extract the path from the request object. 

the get method should handle all inbound requests, concatenate the config.origin.url with whatever comes after the instance url to then hit the origin.

For example, an inbound request to:

https://my-instance-url.com/api/v1/products/xxxx-xxxx-xxxx-xxxx

Should turn around and request the exact same path from the origin url:

https://example.com/api/v1/products/xxxx-xxxx-xxxx-xxxx

**Pull request closed without merging**